financial-templates-lib/UniswapPriceFeed.js

```diff
@@ -0,0 +1,56 @@
+const { delay } = require("./delay");
+const { Logger } = require("./logger/Logger");
+const { LiquidationStatesEnum } = require("../common/Enums");
+
+// A client for getting price information from a uniswap market.
+class UniswapPriceFeed {
+  constructor(abi, web3, uniswapAddress) {
+    this.web3 = web3;
+    this.uniswap = new web3.eth.Contract(abi, uniswapAddress);
+  }
+
+  getCurrentPrice = () => this.currentPrice;
+
+  _update = async () => {
+    // TODO: optimize this call. This may be very slow or break if there are many transactions.
+    const events = await this.uniswap.getPastEvents("Sync", { fromBlock: 0 });
+
+    // If there are no prices, return null and allow
```

Comment:
Did you mean abort instead of allow?

Reply:
Oh no, I just didn't finish writing the comment haha.

Reply:
Done.